builder/pipeline.h

```diff
@@ -19,6 +19,7 @@ struct loop_id {
   const slinky::func* func = nullptr;
   slinky::var var;
 
+  bool innermost(const slinky::func* other) const { return other == func; }
```

Comment:
I'm not sure I understand this. If this loop id has a var defined, isn't this going to return true for the func no matter what, even if that var isn't innermost? Should this be `return other == func && !var.defined()` maybe?

Reply:
The logic I had was that it's impossible to schedule function at it's own loop level, so var doesn't matter, but I agree it's probably good to have it more strict.